ui/src/app/settings/views/Scripts/ScriptsList.test.js

```diff
@@ -0,0 +1,228 @@
+import { mount } from "enzyme";
+import React from "react";
+import { Provider } from "react-redux";
+import { MemoryRouter } from "react-router-dom";
+import configureStore from "redux-mock-store";
+
+import ScriptsList from ".";
+
+const mockStore = configureStore();
+
+describe("ScriptsList", () => {
+  let initialState;
+  beforeEach(() => {
+    initialState = {
+      scripts: {
+        loading: false,
+        loaded: true,
+        items: [
+          {
+            id: 1,
+            name: "commissioning script",
+            description: "a commissioning script",
+            type: 0,
+            history: [
+              {
+                id: 1,
+                comment: "a history item",
+                created: "Tue, 02 Jul 2019 05:24:10 -0000",
+                data: ""
+              }
+            ]
+          },
+          {
+            id: 2,
+            name: "testing script",
+            description: "a testing script",
+            type: 2,
+            history: [
+              {
+                id: 2,
+                comment: "a history item",
+                created: "Tue, 02 Jul 2019 05:24:10 -0000",
+                data: ""
+              }
+            ]
+          },
+          {
+            id: 3,
+            name: "testing script two",
+            description: "another testing script",
+            type: 2,
+            history: [
+              {
+                id: 1,
+                comment: "a history item",
+                created: "Tue, 02 Jul 2019 05:24:10 -0000",
+                data: ""
+              }
+            ]
+          }
+        ]
+      }
+    };
+  });
+
+  it("displays a loading component if loading", () => {
+    const state = { ...initialState };
```

Comment:
I'm not sure it's necessary to shallow copy in these tests as `initialState` is reset between tests anyway.